docs/usage.md

```diff
@@ -101,14 +101,14 @@ In this case, there is no `match` key, so only one rule needs to match:
 `rules` is an array of rules that can be used to target specific files with your defined syntax file.  The rules are processed until the first rule matches, so order your rules in a way that makes sense to you.
 
 ### Filename Rule
-Filename rule defines a filename with regex.  This uses the full file path.
+Filename rule defines a filename with regex.  The pattern is matched against the beginning of the full file path (there is an implicit `^`).
```

Comment:
I mean, i don't think it's confusing or undesirable for it to be matched against the whole path (although i agree the name is non-ideal).

What i meant is that, if you were to drop the `^` from the rule you quoted, it would still behave as if it were there, because the plug-in uses `re.match()`, which always matches against the beginning of the string. This behaviour is sort of uncommon — most tools that do pattern-matching (`grep`, `vim`, Sublime's own find panel) match anywhere in the string by default.

It is true that you *can* explicitly add the `^`, but i don't think that having an explicit anchor is mutually exclusive with having an implicit one. You can add it or not, it won't make any difference. If it seems misleading to you, though, maybe you could say it *acts* like there is an implicit `^`?